### PR TITLE
added scraper

### PR DESCRIPTION
**Pull request by Ellmen**

### CODE REVIEW

File: scraper/igempnp/items.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class TrackItem(scrapy.Item):
+    # define the fields for your item here like:
+    # name = scrapy.Field()
+
+    team_year = scrapy.Field()
+    track = scrapy.Field()
+    region = scrapy.Field()
+    member = scrapy.Field()
+    role = scrapy.Field()
```
Comment (mattsmart) [inline scraper/igempnp/items.py]: need EOF newline

File: scraper/igempnp/items.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import scrapy
+
+
+class MemberItem(scrapy.Item):
+    # define the fields for your item here like:
+    # name = scrapy.Field()
+
+    
+    team_year = scrapy.Field()
```
Comment (mattsmart) [inline scraper/igempnp/items.py:12]: compress these lines too pls

File: scraper/igempnp/items.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    
+
+class TrackItem(scrapy.Item):
+    # define the fields for your item here like:
+    # name = scrapy.Field()
+
+    team_year = scrapy.Field()
```
Comment (mattsmart) [inline scraper/igempnp/items.py:15]: compress these lines too pls

File: scraper/igempnp/pipelines.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    self.exporter.start_exporting()
+
+  def spider_closed(self, spider):
+    self.exporter.finish_exporting()
+    file = self.files.pop(spider)
+    file.close()
+
+  def process_item(self, item, spider):
+    self.exporter.export_item(item)
+    return item
```
Comment (mattsmart) [inline scraper/igempnp/pipelines.py]: need EOF newline and tabs should be 4 spaces not 2!

File: scraper/igempnp/settings.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+
+DUPEFILTER_CLASS = 'scrapy.dupefilter.BaseDupeFilter'
+
+ITEM_PIPELINES = {'igempnp.pipelines.CSVPipeline': 300 }
+
+#DUPEFILTER_DEBUG = True
+
+# Crawl responsibly by identifying yourself (and your website) on the user-agent
+#USER_AGENT = 'igempnp (+http://www.yourdomain.com)'
```
Comment (mattsmart) [inline scraper/igempnp/settings.py:24]: should these settings be set?

File: scraper/igempnp/spiders/MemberSpider.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+class MemberSpider(CrawlSpider):
+
+    dont_filter = True
+
+    name = "track"
+    allowed_domains = ["igem.org"]
+    start_urls = [
+        "http://igem.org/Team_List?year=2007"
+    ]
+
```
Comment (mattsmart) [inline scraper/igempnp/spiders/MemberSpider.py:21]: if this is gonna be length 1 for now, lets keep it as this: 
`start_urls = ["http://igem.org/Team_List?year=2007"]`
if this is gonna be a long list, I think the convention is to do it like this:

```
start_urls = ["http://igem.org/Team_List?year=2007",
              item2,
              etc]
```

File: scraper/igempnp/spiders/MemberSpider.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        #     tablenum += 1
+
+        #     if tablenum == 1:
+
+        #     for thing in sel.xpath('')
+        #     item = MemberItem()
+        #     item['name'] = sel.xpath('text()').extract()
+        #     item['link'] = sel.xpath('@href').extract()
+        #     yield item
+
```
Comment (mattsmart) [inline scraper/igempnp/spiders/MemberSpider.py]: what's this commented block for? keep or no?

File: scraper/igempnp/spiders/__init__.py [hunk 8]
```diff
@@ -0,0 +1,4 @@
+# This package will contain the spiders of your Scrapy project
+#
+# Please refer to the documentation for information on how to create and manage
+# your spiders.
```
Comment (mattsmart) [inline scraper/igempnp/spiders/__init__.py:4]: do we need this init file?

**Comment by mattsmart** [comment]
OK lots of minor style comments, not really any functional ones yet. Can you remove all the .pyc files? I don't think they're necessary and I've added them to our global .gitignore. Also check if the init file is needed.

We should also have more detailed PR descriptions for bigger PRs. Could you update the description at the top of the PR pls to describe how the files interact / what the code does in a few lines?

I'm gonna add a Coding page to the wiki with some python stuff so we can all keep our code consistent. If we end up using any other languages we can add guides there too. I know lots of people on the math team haven't coded in python before, so consistency will help them follow along/learn.

**Comment by Ellmen** [comment]
Sure thing I was just sort of coding it to getting it to work so most of
that is just unfinished/unpolished code. I can add all those fixes when I
get back.

Also, I have no idea if the .pyc files are necessary, I'm pretty sure they
were all automatically generated by scrapy so we might just want to keep
them unless we're really worried about space.

Isaac

**Comment by alexanian** [comment]
Looks good to me! (also Matt, I think, it's hard to understand him through his snickering)

**Pull request merged**